src/object.c

```diff
@@ -776,7 +783,6 @@ size_t objectComputeSize(robj *o, size_t sample_size) {
                 ele = dictGetKey(de);
                 ele2 = dictGetVal(de);
                 elesize += sdsAllocSize(ele) + sdsAllocSize(ele2);
-                elesize += sizeof(struct dictEntry);
```

Comment:
I looked into the whole method again, removal of sizeof(struct dictEntry) is not correct. As the accounting above is only the (dictEntry*) for each case, so it still has to add the size of actual dictEntry. So if you revert that change, only important change left is the list size change. This is essentially the only change required which is done in my pull request #4114.